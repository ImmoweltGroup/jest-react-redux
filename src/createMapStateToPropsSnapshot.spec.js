// @flow

const createMapStateToPropsSnapshot = require('./createMapStateToPropsSnapshot.js');

describe('createMapStateToPropsSnapshot()', () => {
  it('should be a function.', () => {
    expect(typeof createMapStateToPropsSnapshot).toBe('function');
  });

  it('should throw an error if no mapStateToProps function was passed.', () => {
    // $FlowFixMe: suppressing this error since it is a test case
    expect(() => createMapStateToPropsSnapshot({})).toThrow();
  });

  it('should not throw error if no option keys where passed.', () => {
    expect(() =>
      createMapStateToPropsSnapshot({
        mapStateToProps: () => null
      })
    ).not.toThrow();
  });

  it('should mock all selectors and return an output that can be easily compared via Jests snapshot feature.', () => {
    const selectors = {
      getFoo: (state, ownProps) => 'foo'
    };
    const mapStateToProps = (state, ownProps) => ({
      foo: selectors.getFoo(state, ownProps)
    });
    const ownProps = {barOwnProp: 'bar'};
    const results = createMapStateToPropsSnapshot({
      mapStateToProps,
      selectors,
      ownProps,
      selectorImplementationByKey: {
        getFoo: () => 'Some mocked value for getFoo'
      }
    });

    expect(results).toMatchSnapshot();
  });

  it('should support the mapStateToProps to be a curry-function.', () => {
    const selectors = {
      getFoo: (state, ownProps) => 'foo'
    };
    const mapStateToProps = () => (state, ownProps) => ({
      foo: selectors.getFoo(state, ownProps)
    });
    const ownProps = {barOwnProp: 'bar'};
    const results = createMapStateToPropsSnapshot({
      mapStateToProps,
      selectors,
      ownProps,
      selectorImplementationByKey: {
        getFoo: () => 'Some mocked value for getFoo'
      }
    });

    expect(results).toMatchSnapshot();
  });

  it('should support an array of selector maps to be passed.', () => {
    const selectors = [
      {
        getFoo: (state, ownProps) => 'foo'
      }
    ];
    const mapStateToProps = () => (state, ownProps) => ({
      foo: selectors[0].getFoo(state, ownProps)
    });
    const ownProps = {barOwnProp: 'bar'};
    const results = createMapStateToPropsSnapshot({
      mapStateToProps,
      selectors,
      ownProps,
      selectorImplementationByKey: {
        getFoo: () => 'Some mocked value for getFoo'
      }
    });

    expect(results).toMatchSnapshot();
  });

  it('should support custom state to be used from the options.', () => {
    const selectors = [
      {
        getFoo: (state, ownProps) => 'foo'
      }
    ];
    const mapStateToProps = () => (state, ownProps) => ({
      foo: selectors[0].getFoo(state, ownProps)
    });
    const ownProps = {barOwnProp: 'bar'};
    const state = {barState: true};
    const results = createMapStateToPropsSnapshot({
      mapStateToProps,
      selectors,
      ownProps,
      state,
      selectorImplementationByKey: {
        getFoo: () => 'Some mocked value for getFoo'
      }
    });

    expect(results).toMatchSnapshot();
  });
});
