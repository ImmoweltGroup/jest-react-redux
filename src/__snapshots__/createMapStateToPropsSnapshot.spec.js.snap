// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMapStateToPropsSnapshot() should mock all selectors and return an output that can be easily compared via Jests snapshot feature. 1`] = `
Object {
  "calls": Object {
    "getFoo": Object {
      "argsByCallCount": Object {
        "0": Array [
          Object {
            "__mockState": true,
          },
          Object {
            "barOwnProp": "bar",
          },
        ],
      },
      "callCount": 1,
    },
  },
  "stateProps": Object {
    "foo": "Some mocked value for getFoo",
  },
}
`;

exports[`createMapStateToPropsSnapshot() should support an array of selector maps to be passed. 1`] = `
Object {
  "calls": Object {
    "getFoo": Object {
      "argsByCallCount": Object {
        "0": Array [
          Object {
            "__mockState": true,
          },
          Object {
            "barOwnProp": "bar",
          },
        ],
      },
      "callCount": 1,
    },
  },
  "stateProps": Object {
    "foo": "Some mocked value for getFoo",
  },
}
`;

exports[`createMapStateToPropsSnapshot() should support custom state to be used from the options. 1`] = `
Object {
  "calls": Object {
    "getFoo": Object {
      "argsByCallCount": Object {
        "0": Array [
          Object {
            "barState": true,
          },
          Object {
            "barOwnProp": "bar",
          },
        ],
      },
      "callCount": 1,
    },
  },
  "stateProps": Object {
    "foo": "Some mocked value for getFoo",
  },
}
`;

exports[`createMapStateToPropsSnapshot() should support the mapStateToProps to be a curry-function. 1`] = `
Object {
  "calls": Object {
    "getFoo": Object {
      "argsByCallCount": Object {
        "0": Array [
          Object {
            "__mockState": true,
          },
          Object {
            "barOwnProp": "bar",
          },
        ],
      },
      "callCount": 1,
    },
  },
  "stateProps": Object {
    "foo": "Some mocked value for getFoo",
  },
}
`;
